Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -517,6 +519,52 @@ def check_torrents_channel(src, defer_param=None):
         d.addCallback(check_torrents_channel)
         return d
 
+    @deferred(timeout=20)
+    def test_chn_native_load(self):
+        self.session.get_dispersy = lambda: True
+        self.session.lm.dispersy = Dispersy(ManualEnpoint(0), self.getStateDir())
+        dispersy_cid_hex = "abcd" * 9 + "0012"
+        dispersy_cid = binascii.unhexlify(dispersy_cid_hex)
+
+        # create channel and insert torrent
+        self.create_fake_allchannel_community()
+        self.create_torrents_in_channel(dispersy_cid_hex)
+
+        self.session.download_torrentfile = \
+            lambda dummy_ihash, function, _: function(binascii.hexlify(TORRENT_FILE_INFOHASH))
+
+        def get_bin_torrent(_):
+            """
+            get binary data of a torrent
+            """
+            f = open(TORRENT_FILE, "rb")
+            bdata = f.read()
+            f.close()
+            return bdata
+
+        self.session.get_collected_torrent = get_bin_torrent
+
+        self.boosting_manager.add_source(dispersy_cid)
+
+        def check_loaded(src, defer_param=None):
+            """
+            check if a torrent has been loaded
+            """
+            if defer_param is None:
```

Comment:
Please use the looping call mechanism here that I recently adopted in your tests (see the other tests in this file to see how I did that).